Packages/com.unity.input/Runtime/Input.cs

```diff
@@ -647,27 +648,125 @@ public static bool GetKeyUp(KeyCode key)
+        private static (float radius, float variance) RadiusValues(Vector2 vRadius)
         {
-            ////TODO: implement properly
-            return GetAxis(axisName);
+            var x = vRadius.x;
+            var y = vRadius.y;
+            var max = Mathf.Max(x, y);
+            var min = Mathf.Min(x, y);
+            var variance = (max - min) / 2;
+            var radius = min + variance;
+
+            return (radius, variance);
+        }
+
+        private static (float altitudeAngle, float azimuthAngle) TiltValues(Vector2 vTilt)
+        {
+            var azimuthAngle = Mathf.Acos(vTilt.x);
+            var altitudeAngle = Mathf.Acos(vTilt.y);
+            return (altitudeAngle, azimuthAngle);
+        }
+
+        public static Touch ToTouch(InputSystem.Controls.TouchControl control)
+        {
+            Touch touch = new Touch();
+            touch.fingerId = control.touchId.ReadValue();
+            touch.position = control.position.ReadValue();
+            touch.rawPosition = control.startPosition.ReadValue();
+            touch.deltaPosition = control.delta.ReadValue();
+            //touch.deltaTime = TODO: How to implement this? Device has lastUpdateTime() but only for single touch
+            touch.tapCount = control.tapCount.ReadValue();
+            touch.phase = ToLegacy(control.phase.ReadValue());
+            touch.pressure = control.pressure.ReadValue();
+            touch.maximumPossiblePressure = 1.0f;
+            touch.type = control.indirectTouch.IsPressed() ? TouchType.Indirect : TouchType.Direct;
+            touch.altitudeAngle = 0;  // TODO: Not available in TouchControl
+            touch.azimuthAngle = 0;   // TODO: Not available in TouchControl
+
+            var(radius, variance) = RadiusValues(control.radius.ReadValue());
+            touch.radius = radius;
+            touch.radiusVariance = variance;
+
+            return touch;
         }
 
         public static Touch GetTouch(int index)
         {
-            return default;
+            if (index >= touchCount)
+                throw new IndexOutOfRangeException();
+
+            return ToTouch(Touchscreen.current.touches[index]);
+        }
+
+        public static int touchCount
+        {
+            get
+            {
+                var count = 0;
+                if (Touchscreen.current != null)
+                {
+                    foreach (var touch in Touchscreen.current.touches)
+                    {
+                        if (touch.phase.ReadValue() != InputSystem.TouchPhase.None)
+                            count++;
+                    }
+                }
+
+                return count;
+            }
+        }
+        public static Touch[] touches
+        {
+            get
+            {
+                var result = new List<Touch>();
+                if (Touchscreen.current != null)
+                {
+                    var convertedTouches = from t in Touchscreen.current.touches
```

Comment:
This needs to use the `EnhancedTouch` API. Polling `Touchscreen` directly will not give the right results. It'll drop touches and lead to incorrect touch sequences.